java-frontend/src/main/java/org/sonar/plugins/java/api/tree/TryStatementTree.java

```diff
@@ -45,10 +45,17 @@
   SyntaxToken openParenToken();
 
   /**
+   * Java 9 allows resources to be declared by existing effectively final variables.
+   * This method will not return trees of resources declared by such variables, only resources declared with new variables will be returned (as it was until Java 9).
+   * Use {@link #resourceList()} to get all declared resources.
+   *
    * @since Java 1.7
    */
+  @Deprecated
   ListTree<VariableTree> resources();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)